lib/ddr/models/has_children.rb

```diff
@@ -4,8 +4,20 @@ module HasChildren
 
       def first_child
         child_class = self.class.reflect_on_association(:children).klass
-        solr_field = child_class.reflect_on_association(:parent).solr_key
-        child_class.where(solr_field => id).order("#{Ddr::Index::Fields::LOCAL_ID} ASC").first
+        if has_struct_metadata?
+          child_class.find(first_id_from_struct_map)
+        else
+          solr_field = child_class.reflect_on_association(:parent).solr_key
+          child_class.where(solr_field => id).order("#{Ddr::Index::Fields::LOCAL_ID} ASC").first
+        end
+      end
+
+      private
+
+      def first_id_from_struct_map
```

Comment:
I have a mild feeling that this (or part of this) method belongs in the `Structure` class.